libs/langchain_v1/langchain/agents/middleware/human_in_the_loop.py

```diff
@@ -1,135 +1,201 @@
 """Human in the loop middleware."""
 
-from typing import Any
-
-from langgraph.prebuilt.interrupt import (
-    ActionRequest,
-    HumanInterrupt,
-    HumanInterruptConfig,
-    HumanResponse,
-)
+from typing import Any, Literal
+
+from langchain_core.messages import AIMessage, ToolCall, ToolMessage
 from langgraph.types import interrupt
+from typing_extensions import NotRequired, TypedDict
 
-from langchain.agents.middleware._utils import _generate_correction_tool_messages
 from langchain.agents.middleware.types import AgentMiddleware, AgentState
 
-ToolInterruptConfig = dict[str, HumanInterruptConfig]
+
+class ActionRequest(TypedDict):
```

Comment:
I don't think `ActionRequest` works well in terms of communicating what the purpose is. `HumanInterrupt` actually made is much simpler to understand that something was being sent for review and involving interrupting the workflow of the graph.

Any alternatives?